### PR TITLE
Add Plywrks Lune

### PR DESCRIPTION
**Pull request by ramonimbao**
## Description

Add the Plywrks Lune.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/plywrks/lune/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+
+char layer[16];
```
Comment (drashna) [inline keyboards/plywrks/lune/keymaps/default/keymap.c:19]:
```suggestion
```

File: keyboards/plywrks/lune/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                 _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                 _______,          _______, _______, _______,          _______, _______,                                              _______
+    )
+};
+
+#ifdef OLED_ENABLE
+bool oled_task_user(void) {
+    // Host Keyboard Layer Status
+    sprintf(layer, "Layer :%u", get_highest_layer(layer_state));
+    oled_write(layer, false);
```
Comment (drashna) [inline keyboards/plywrks/lune/keymaps/default/keymap.c:42]: This saves you ~1.5k on firmware space. 
```c
    oled_write_P(PSTR("Layer :", false);
    oled_write(get_u8_str(get_highest_layer(layer_state), ' '), false);
```

File: keyboards/plywrks/lune/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+
+char layer[16];
```
Comment (drashna) [inline keyboards/plywrks/lune/keymaps/via/keymap.c:19]:
```suggestion
```

File: keyboards/plywrks/lune/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_set_cursor(0, 1);
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+
+    return false;
+}
+#endif
```
Comment (drashna) [inline keyboards/plywrks/lune/keymaps/default/keymap.c:53]:
```suggetion
```

File: keyboards/plywrks/lune/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_set_cursor(0, 1);
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+
+    return false;
+}
+#endif
```
Comment (drashna) [inline keyboards/plywrks/lune/keymaps/via/keymap.c:53]:
```suggestion
```

File: keyboards/plywrks/lune/lune.c [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "lune.h"
```
Comment (drashna) [inline keyboards/plywrks/lune/lune.c:17]:
```suggestion
#include "lune.h"

#ifdef OLED_ENABLE
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
    // Host Keyboard Layer Status
    oled_write_P(PSTR("Layer :", false);
    oled_write(get_u8_str(get_highest_layer(layer_state), ' '), false);
    
    oled_advance_page(true);

        // Host Keyboard LED Status
    led_t led_state = host_keyboard_led_state();
    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);

    return false;
}
#endif
```
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/plywrks/lune/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                 KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+                 KC_LCTL,          KC_LALT, KC_SPC,  MO(1),            KC_SPC,  KC_RALT,                                              KC_RCTL
+    ),
+    [1] = LAYOUT_all(
+        RESET,   _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,   KC_DEL,
+        KC_HOME, _______, _______, KC_UP,   _______, _______, _______, RGB_HUD, RGB_HUI, RGB_SAD, RGB_VAD, RGB_VAI, _______, _______, _______,
+        KC_END,  _______, KC_LEFT, KC_DOWN, KC_RGHT, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                 _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                 _______,          _______, _______, _______,          _______, _______,                                              _______
+    )
```
Comment (fauxpark) [inline keyboards/plywrks/lune/keymaps/via/keymap.c:32]: Need two more layers, or a config.h with `#define DYNAMIC_KEYMAP_LAYER_COUNT 2`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/plywrks/lune/keymaps/via/keymap.c (hunk 7 above)
Comment (ramonimbao) [reply keyboards/plywrks/lune/keymaps/via/keymap.c:46]: Thanks! Added the two layers.

File: keyboards/plywrks/lune/info.json [hunk 8]
```diff
@@ -0,0 +1,75 @@
+{
+    "keyboard_name": "Lune",
+    "url": "",
+    "maintainer": "Ramon Imbao",
```
Comment (noroadsleft) [inline keyboards/plywrks/lune/info.json:4]: This field is meant to reference your GitHub username.

```suggestion
    "maintainer": "ramonimbao",
```

File: keyboards/plywrks/lune/info.json [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+                {"x":0.25, "y":1},
+                {"x":1.5, "y":1, "w":1.5},
+                {"x":3, "y":1}, {"x":4, "y":1},
+                {"x":5, "y":1}, {"x":6, "y":1},
+                {"x":7, "y":1}, {"x":9.5, "y":1},
+                {"x":10.5, "y":1}, {"x":11.5, "y":1},
+                {"x":12.5, "y":1}, {"x":13.5, "y":1},
+                {"x":14.5, "y":1}, {"x":15.5, "y":1},
+                {"x":16.5, "y":1, "w":1.5},
```
Comment (noroadsleft) [inline keyboards/plywrks/lune/info.json:33]: Strange formatting here.

```suggestion
                {"x":0.25, "y":1},
                {"x":1.5, "y":1, "w":1.5},
                {"x":3, "y":1},
                {"x":4, "y":1},
                {"x":5, "y":1},
                {"x":6, "y":1},
                {"x":7, "y":1},
                {"x":9.5, "y":1},
                {"x":10.5, "y":1},
                {"x":11.5, "y":1},
                {"x":12.5, "y":1},
                {"x":13.5, "y":1},
                {"x":14.5, "y":1},
                {"x":15.5, "y":1},
                {"x":16.5, "y":1, "w":1.5},
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Also a general comment:

`LAYOUT_all` appears to be compatible with - if not entirely identical to - `LAYOUT_alice_split_bs`. You could replace all the instances of the former with the latter, and make the board compatible with the `alice_split_bs` community layout.

File: keyboards/plywrks/lune/info.json [hunk 10]
```diff
@@ -0,0 +1,81 @@
+{
+    "keyboard_name": "Lune",
+    "url": "",
+    "maintainer": "ramonimbao",
```
Comment (fauxpark) [inline keyboards/plywrks/lune/info.json:4]:
```suggestion
    "keyboard_name": "Lune",
    "manufacturer": "plywrks",
    "url": "",
    "maintainer": "ramonimbao",
    "usb": {
        "vid": "0x706C",
        "pid": "0x7901",
        "device_version": "0.0.1"
    },
```

File: keyboards/plywrks/lune/config.h [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x706C // P L
+#define PRODUCT_ID   0x7901 // Y 1
+#define DEVICE_VER   0x0001
+#define MANUFACTURER plywrks
+#define PRODUCT      Lune
+
```
Comment (fauxpark) [inline keyboards/plywrks/lune/config.h:28]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**